metals/src/main/scala/scala/meta/internal/metals/codeactions/BracelessBracefulSwitchCodeAction.scala

```diff
@@ -0,0 +1,558 @@
+        trees
+          .findLastEnclosingAt[Tree](
+            path,
+            range.getStart(),
+            applyWithSingleFunction
+          )
+      else None
+
+    val result =
+      for {
+        tree <- maybeTree
+        document <- buffers.get(path)
+      } yield {
+        tree match {
+          case _: Pkg => None.toSeq
+          case classDefn: Defn.Class =>
+            createCodeActionForTemplateHolder(
+              classDefn,
+              path,
+              document,
+              classDefn.templ,
+              "class definition"
+            ).toSeq
+          case objectDefn: Defn.Object =>
+            createCodeActionForTemplateHolder(
+              objectDefn,
+              path,
+              document,
+              objectDefn.templ,
+              "object definition"
+            ).toSeq
+          case traitDefn: Defn.Trait =>
+            createCodeActionForTemplateHolder(
+              traitDefn,
+              path,
+              document,
+              traitDefn.templ,
+              "trait definition"
+            ).toSeq
+          case enumDefn: Defn.Enum =>
+            createCodeActionForTemplateHolder(
+              enumDefn,
+              path,
+              document,
+              enumDefn.templ,
+              "enum definition"
+            ).toSeq
+          case valDefn: Defn.Val =>
+            createCodeActionForPotentialBlockHolder(
+              valDefn,
+              path,
+              document,
+              valDefn.rhs,
+              "val definition"
+            ).toSeq
+          case varDefn: Defn.Var =>
+            varDefn.rhs.flatMap {
+              createCodeActionForPotentialBlockHolder(
+                varDefn,
+                path,
+                document,
+                _,
+                "var definition"
+              )
+            }.toSeq
+          case defDefn: Defn.Def =>
+            createCodeActionForPotentialBlockHolder(
+              defDefn,
+              path,
+              document,
+              defDefn.body,
+              "def definition"
+            ).toSeq
+          case termTry: Term.Try =>
+            Seq(
+              createCodeActionForPotentialBlockHolder(
+                termTry.expr,
+                path,
+                document,
+                termTry.expr,
+                "try expression"
+              ),
+              createCodeActionForCatchP(
+                termTry,
+                path,
+                document,
+                "catch expression"
+              ),
+              termTry.finallyp.flatMap(finallyp =>
+                createCodeActionForPotentialBlockHolder(
+                  finallyp,
+                  path,
+                  document,
+                  finallyp,
+                  "finally expression"
+                )
+              )
+            ).flatten
+          case caseTree: Case =>
+            createCodeActionForCasesListWrapper(caseTree, path).toSeq
```

Comment:
We should only show dropping braces in this case. Though it would be awesome to show it based on some linting information about redundant braces. This could be done via scalafix.

Anyone, for this I would only show an action to drop braces, but not to add them

Reply:
Oh, this one is a TODO